### PR TITLE
Add proxy importing, fix proxy issues (see desc.) - closes #1124

### LINKED ISSUE #1124: [Suggestion] Alt list/Proxy List
Add a feature to import proxies as a ip:port list, and alts as a username:password list.

Considering we have things like the swarm module, it's kind of absurd that we have to manually input proxies and alts.

### PR DESCRIPTION
**Pull request by SIMULATAN**
- fix proxy selection bug
- fix ability to add proxies with 0 as the port or no address
- fix showing empty proxy names

Damn, that's a huge update!

### CODE REVIEW

File: src/main/java/meteordevelopment/meteorclient/utils/render/color/Color.java [hunk 1]
```diff
@@ -10,7 +10,36 @@
 import net.minecraft.nbt.NbtCompound;
 import net.minecraft.util.math.Vec3d;
 
+@SuppressWarnings("unused")
 public class Color implements ICopyable<Color>, ISerializable<Color> {
+
+    public static final Color white = new Color(java.awt.Color.WHITE);
+    public static final Color WHITE = white;
+    public static final Color lightGray = new Color(java.awt.Color.LIGHT_GRAY);
+    public static final Color LIGHT_GRAY = lightGray;
```
Comment (C10udburst) [inline src/main/java/meteordevelopment/meteorclient/utils/render/color/Color.java:19]: is there a reason why there are two variables for each color?

Comment (SIMULATAN) [reply src/main/java/meteordevelopment/meteorclient/utils/render/color/Color.java:19]: It's to follow along the java.awt format

Comment (MineGame159) [reply src/main/java/meteordevelopment/meteorclient/utils/render/color/Color.java:19]: Leave just the uppercase ones.

Comment (SIMULATAN) [reply src/main/java/meteordevelopment/meteorclient/utils/render/color/Color.java]: Done in e6dd40f

Comment (SIMULATAN) [reply src/main/java/meteordevelopment/meteorclient/utils/render/color/Color.java]: If you're fine with how it looks, just squash merge it

File: src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                );
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            add(theme.label("Invalid File!"));
+        }
+        add(theme.horizontalSeparator()).expandX();
+        WButton btnBack = add(theme.button("Back")).expandX().widget();
+        btnBack.action = () -> MeteorClient.mc.setScreen(parent);
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java:80]: While this works to close the screen it is better practise to call the `onClose()` method.

Sorry for not pointing it out earlier.

**Review by MineGame159 — CHANGES_REQUESTED** [review]

File: src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java (hunk 2 above)
Comment (SIMULATAN) [reply src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java:80]: `onClose` sounded like the close listener which is why I didn't use it. Also, there are other parts in the code where I found this method of switching screens which is why I used it.

Comment (MineGame159) [reply src/main/java/meteordevelopment/meteorclient/gui/screens/ProxiesImportScreen.java]: Oh

**Comment by SIMULATAN** [comment]
thanks mimgam, very cool

**Pull request merged**